front/src/applications/stdcm/hooks/useStdcmResults.ts

```diff
@@ -0,0 +1,203 @@
+
+  const { data: otherSelectedTrainSchedule } =
+    osrdEditoastApi.endpoints.getV2TrainScheduleById.useQuery(
+      {
+        id: selectedTrainId as number,
+      },
+      { skip: !selectedTrainId }
+    );
+
+  const { data: timetable } = osrdEditoastApi.endpoints.getV2TimetableById.useQuery(
+    { id: timetableId! },
+    {
+      skip: !timetableId,
+    }
+  );
+
+  const selectedTrainSchedule = useMemo(
+    () =>
+      selectedTrainId !== STDCM_TRAIN_ID && otherSelectedTrainSchedule
+        ? otherSelectedTrainSchedule
+        : stdcmTrainResult,
+    [selectedTrainId, stdcmTrainResult, otherSelectedTrainSchedule]
+  );
+
+  useEffect(() => {
+    const getPathProperties = async (_infraId: number, path: PathfindingResultSuccess) => {
+      const pathPropertiesParams: PostV2InfraByInfraIdPathPropertiesApiArg = {
+        infraId: _infraId,
+        props: ['electrifications', 'geometry', 'operational_points'],
+        pathPropertiesInput: {
+          track_section_ranges: path.track_section_ranges,
+        },
+      };
+      const { geometry, operational_points, electrifications } =
+        await postPathProperties(pathPropertiesParams).unwrap();
+
+      if (geometry && operational_points && electrifications) {
+        const pathStepsWihPosition = compact(pathSteps).map((step, i) => ({
+          ...step,
+          positionOnPath: path.path_items_positions[i],
+        }));
+
+        const suggestedOperationalPoints: SuggestedOP[] = formatSuggestedOperationalPoints(
+          operational_points,
+          geometry,
+          path.length
+        );
+
+        const updatedSuggestedOPs = upsertViasInOPs(
+          suggestedOperationalPoints,
+          pathStepsWihPosition
+        );
+
+        setPathProperties({
+          electrifications,
+          geometry,
+          suggestedOperationalPoints: updatedSuggestedOPs,
+          allVias: updatedSuggestedOPs,
+          length: path.length,
+        });
+      }
+    };
+
+    if (infraId && stdcmResponse) {
+      const { path } = stdcmResponse;
+      getPathProperties(infraId, path);
+    }
+  }, [stdcmResponse]);
+
+  useEffect(() => {
+    const getSpaceTimeChartData = async (
+      _infraId: number,
+      path: PathfindingResultSuccess,
+      trainSchedulesIds: number[],
+      stdcmSimulation: StdcmV2SuccessResponse['simulation'],
+      stdcmDepartureTime: string,
+      stdcmRollingStockLength: number
+    ) => {
+      const { blocks, routes, track_section_ranges } = path;
+      const projectPathTrainResult = await projectTrainSchedules({
+        infra: _infraId,
+        ids: trainSchedulesIds,
+        projectPathInput: { blocks, routes, track_section_ranges },
+      }).unwrap();
+
+      const newSpaceTimeData = Object.keys(projectPathTrainResult).reduce((result, trainId) => {
+        const currentProjectedTrain = projectPathTrainResult[trainId];
+
+        // const matchingTrain = trainSchedules.find((train) => train.id === +trainId);
```

Comment:
It's an oversight or a future update?
We could put a to do  so we don't forget it.

Reply:
Yes nice catch, I fixed this in a fixup commit ;)